examples/spin-wagi-http/http-rust/src/lib.rs

```diff
@@ -1,15 +1,12 @@
 use anyhow::Result;
-use spin_sdk::{
-    http::{Request, Response},
-    http_component,
-};
+use spin_sdk::http_component;
 
 /// A simple Spin HTTP component.
 #[http_component]
-fn goodbye_world(req: Request) -> Result<Response> {
+fn goodbye_world(req: http::Request<()>) -> Result<http::Response<&'static str>> {
```

Comment:
Again, that's because the previous example was an alias for `Request<Option<bytes::Bytes>>` so while it looked simpler at first glance, I'd argue it wasn't (especially considering the `bytes::Bytes` type is actually really user unfriendly IMO. We can offer type aliases if we think that that would be helpful, but I want to avoid falling into the trap of thinking that this diff which superficially sees the edition of more characters is necessarily more complicated when I'd argue it's not.

Reply:
As the person who initially made the change to have bodies be `Option<Bytes>`, that has proven to be really inflexible and not in line with what a Rust developer would expect when writing a handler in a web context.

We can consider having aliases for special (maybe common?) scenarios like request bodies.

Finally, perhaps it would be easier for the example just to return a `Response<String>` here?

Reply:
Going to merge this PR first, and then we can take another closer look at the exact set of examples in another PR.